### DISCUSSION TITLE
UI Issues in Latest <model-viewer> for Android (v3.5.0)

### DISCUSSION THREAD

**Discussion by adityakotha03** [google/model-viewer#4840] [Q&A]
### Description

In the latest version of <model-viewer> for Android, there are several UI elements that do not display at all, like the ones that tell users to point the camera at an empty space or warn them when it is too dark. These issues are exclusive to the latest version and were not present in previous versions.

These Elements are missing: 
![UI](https://github.com/user-attachments/assets/3d6d69e3-b8ab-4d5e-b20c-bc935c009184)

#### Live Demo
https://modelviewer.dev/examples/augmentedreality/#ar

### Version

- model-viewer: v3.5.0

**Comment by elalish** [comment]
Can you please also show an image of the current experience that's not working, so we can see the difference? Your current image is of Scene Viewer, which is an Android system app and not part of model-viewer. Our default AR experience on Android is WebXR, which doesn't have any strings and such, but you can add your own with HTML, as in our [example](https://modelviewer.dev/examples/augmentedreality/index.html#webXR). This also allows you to customize, localize, etc. 

Nothing much has changed around AR between 3.4 and 3.5 - are you sure their behavior is different? Scene Viewer itself is more likely to have changed independently.

**Comment by adityakotha03** [comment]
Hey thanks for the quick reply here is the current UI compared to previous models. That's interesting because using different versions of model view I get different Android system screen, which in my opinion seems wired.
![Screenshot_20240722-215232](https://github.com/user-attachments/assets/a33bb170-1668-4e21-8e0c-ec3e4c0583d6)

**Reply by elalish** [reply]
That is strange indeed, and not my experience. Can you share what device models and browsers you have tested this on? And can you share two links, one for the v3.4 repro and one for the v3.5 repro?

**Reply by elalish** [reply]
Most often you end up with Scene Viewer instead of WebXR because you either refused an XR permission prompt, or served the page over HTTP instead of HTTPS.

**Reply by adityakotha03** [reply]
Hey I am testing this on Pixel 8 Pro, and my chorme version is 126.0.6478.123. https://modelviewer.dev/examples/augmentedreality/#ar 
This is the model viewer example for v3.5 

And this example uses an old version of model viewer, looks like it's V1.5 and which gives me the normal screen which I used to get when using v3.4
https://react-model-viewer.netlify.app/

**Reply by adityakotha03** [reply]
> Most often you end up with Scene Viewer instead of WebXR because you either refused an XR permission prompt, or served the page over HTTP instead of HTTPS.

But this wouldn't be the case when I am able to use one model with the screen I shared in the beginning and the second with a different. Also I would like to mention that model placement works fine, no issues in both cases but it's just the UI elements are different.

**Reply by elalish** [reply]
It's in fact a completely different app that's running, it just looks somewhat similar. In WebXR mode you're still in the browser, so you can still run your own JS and get events and such. Scene Viewer is a completely separate app that you as a developer can't interact with. I believe v1.5 was from so many years ago that WebXR didn't exist at the time. It's now our default because it enables so many things that are impossible with Scene Viewer, e.g. these examples in AR: https://modelviewer.dev/examples/scenegraph/

**Reply by adityakotha03** [reply]
Thank you so much for taking your time in helping me, you were right the V1.5 was using scene viewer hence it was triggering scene viewer app, and in v3.5 if we only set it to use Scene-Viewer I get back the same UI. This was caused due to my misunderstanding on webxr and scene viewer. Thanks a lot for helping me out.
